esi/src/lib.rs

```diff
@@ -33,7 +36,10 @@ pub struct Processor {
 }
 
 impl Processor {
-    pub fn new(original_request_metadata: Option<Request>, configuration: Configuration) -> Self {
+    pub const fn new(
```

Comment:
I wasn't aware of the nuance of this but had a look at the Rust docs and it makes a lot of sense. Thanks.